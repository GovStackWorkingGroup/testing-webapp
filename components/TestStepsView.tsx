import React, { useCallback } from 'react';
import { useIntl } from 'react-intl';
import { RiCloseCircleLine, RiCheckboxCircleLine } from 'react-icons/ri';
import { TestsDetailsType } from '../service/types';

type Props = {
  testDetails: TestsDetailsType;
};

const TestStepsView = ({ testDetails }: Props) => {
  const { formatMessage } = useIntl();
  const format = useCallback(
    (id: string) => formatMessage({ id }),
    [formatMessage]
  );

  return (
    <div className='test-steps-container'>
      <p className='test-steps-scenario-label' data-testid='test-steps-scenario-label'>
        {format('app.scenario.label')}
      </p>
      <p className='test-steps-scenario-title' data-testid='test-steps-scenario-title'>{testDetails.scenario}</p>
      {testDetails.steps.map((step, idx) => (
        <div className='test-steps-result' key={`test-step-${idx}`}>
          <div className='test-steps-icon'>
            {step.result ? (
              <RiCheckboxCircleLine className='test-steps-passed-icon' />
            ) : (
              <RiCloseCircleLine className='test-steps-failed-icon' />
            )}
          </div>
          <div className='test-steps-description'>
            <p>{`${step.type} ${step.text}`}</p>
          </div>
        </div>
      ))}
    </div>
  );
};

export default TestStepsView;
