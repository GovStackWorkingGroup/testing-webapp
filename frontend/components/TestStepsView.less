@import '../styles/Variables.less';

.test-steps-container {
  border: 1px solid #dfdfea;
  border-radius: 8px;
  margin: 20px 24px 20px 16px;
  padding: 24px 16px;
}

.test-steps-scenario-label {
  font-size: 12px;
  color: @text-secondary;
}

.test-steps-scenario-title {
  font-style: italic;
  padding: 10px 0;
}

.test-steps-result {
  display: flex;
  gap: 8px;
  margin: 16px 0;
}

.test-steps-result-description {
  display: flex;
  gap: 5px;
  font-size: 16px;
}

.test-steps-icon {
  width: 14px;
}

.test-steps-passed-icon {
  color: @success;
  font-size: 16px;
}

.test-steps-failed-icon {
  color: @fail;
  font-size: 16px;
}
