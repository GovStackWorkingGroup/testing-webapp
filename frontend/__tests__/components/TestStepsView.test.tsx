import TestStepsView from '../../components/TestStepsView';
import { render } from '../test-utils/test-utils';
import {
  testFailedDetails,
  testPassedDetails,
} from './table/mockedData/ProductTable';

describe('Unit tests for TestStepsView component:', () => {
  const SCENARIO_LABEL_TEST_ID = 'test-steps-scenario-label';
  const SCENARIO_TITLE_TEST_ID = 'test-steps-scenario-title';

  it('should render the component when tests passed and match snapshot', () => {
    const { container } = render(
      <TestStepsView testDetails={testPassedDetails} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the component when tests failed and match snapshot', () => {
    const { container } = render(
      <TestStepsView testDetails={testFailedDetails} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should have corresponding values', () => {
    const { getByTestId } = render(
      <TestStepsView testDetails={testPassedDetails} />
    );

    expect(getByTestId(SCENARIO_LABEL_TEST_ID)).toHaveTextContent('Scenario:');
    expect(getByTestId(SCENARIO_TITLE_TEST_ID)).toHaveTextContent(
      'The user successfully creates record "John Smith" in the Digital Registries database.'
    );
  });
});
