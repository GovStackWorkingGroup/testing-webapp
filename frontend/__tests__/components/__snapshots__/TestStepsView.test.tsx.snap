// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for TestStepsView component: should render the component when tests failed and match snapshot 1`] = `
<div>
  <div
    class="test-steps-container"
  >
    <p
      class="test-steps-scenario-label"
      data-testid="test-steps-scenario-label"
    >
      Scenario:
    </p>
    <p
      class="test-steps-scenario-title"
      data-testid="test-steps-scenario-title"
    >
      The user is not able to create a record in the database which not exist
    </p>
    <div
      class="test-steps-result"
    >
      <div
        class="test-steps-icon"
      >
        <svg
          class="test-steps-failed-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm0-9.414l2.828-2.829 1.415 1.415L13.414 12l2.829 2.828-1.415 1.415L12 13.414l-2.828 2.829-1.415-1.415L10.586 12 7.757 9.172l1.415-1.415L12 10.586z"
            />
          </g>
        </svg>
      </div>
      <div
        class="test-steps-description"
      >
        <p>
          Given The user wants to create a new record "Anna Stock" in the Digital Registries database
        </p>
      </div>
    </div>
    <div
      class="test-steps-result"
    >
      <div
        class="test-steps-icon"
      >
        <svg
          class="test-steps-failed-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm0-9.414l2.828-2.829 1.415 1.415L13.414 12l2.829 2.828-1.415 1.415L12 13.414l-2.828 2.829-1.415-1.415L10.586 12 7.757 9.172l1.415-1.415L12 10.586z"
            />
          </g>
        </svg>
      </div>
      <div
        class="test-steps-description"
      >
        <p>
          When The user triggers an action to create a new record "Anna Stock" in the database
        </p>
      </div>
    </div>
    <div
      class="test-steps-result"
    >
      <div
        class="test-steps-icon"
      >
        <svg
          class="test-steps-failed-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm0-9.414l2.828-2.829 1.415 1.415L13.414 12l2.829 2.828-1.415 1.415L12 13.414l-2.828 2.829-1.415-1.415L10.586 12 7.757 9.172l1.415-1.415L12 10.586z"
            />
          </g>
        </svg>
      </div>
      <div
        class="test-steps-description"
      >
        <p>
          Then Operation results in an error for create "Anna Stock"
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for TestStepsView component: should render the component when tests passed and match snapshot 1`] = `
<div>
  <div
    class="test-steps-container"
  >
    <p
      class="test-steps-scenario-label"
      data-testid="test-steps-scenario-label"
    >
      Scenario:
    </p>
    <p
      class="test-steps-scenario-title"
      data-testid="test-steps-scenario-title"
    >
      The user successfully creates record "John Smith" in the Digital Registries database.
    </p>
    <div
      class="test-steps-result"
    >
      <div
        class="test-steps-icon"
      >
        <svg
          class="test-steps-passed-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-.997-4L6.76 11.757l1.414-1.414 2.829 2.829 5.656-5.657 1.415 1.414L11.003 16z"
            />
          </g>
        </svg>
      </div>
      <div
        class="test-steps-description"
      >
        <p>
          Given The user wants to create a new record "John Smith" in the Digital Registries database
        </p>
      </div>
    </div>
    <div
      class="test-steps-result"
    >
      <div
        class="test-steps-icon"
      >
        <svg
          class="test-steps-passed-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-.997-4L6.76 11.757l1.414-1.414 2.829 2.829 5.656-5.657 1.415 1.414L11.003 16z"
            />
          </g>
        </svg>
      </div>
      <div
        class="test-steps-description"
      >
        <p>
          When The user triggers an action to create a new record "John Smith" in the database
        </p>
      </div>
    </div>
    <div
      class="test-steps-result"
    >
      <div
        class="test-steps-icon"
      >
        <svg
          class="test-steps-passed-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-.997-4L6.76 11.757l1.414-1.414 2.829 2.829 5.656-5.657 1.415 1.414L11.003 16z"
            />
          </g>
        </svg>
      </div>
      <div
        class="test-steps-description"
      >
        <p>
          Then Operation finishes successfully for create user "John Smith"
        </p>
      </div>
    </div>
  </div>
</div>
`;
